TypeApplications interferes with pattern bindings with spaces

Below is a valid Haskell program without `-XTypeApplications`:

```haskell
main = case [1] of
  xs @ (x:_) -> print (x, xs)
```

However, it is rejected by Ormolu since it causes a parse error with `TypeApplications` which is enabled by default:

```
$ runhaskell test.hs
(1,[1])

$ runhaskell -XTypeApplications test.hs
test.hs:2:3: error: Parse error in pattern: xs @(x : _)
  |
2 |   xs @ (x:_) -> print (x, xs)
  |   ^^^^^^^^^^
```

We probably should not enable TypeApplications by default.

--- a/data/examples/declaration/value/function/pattern/as-pattern-out.hs
+++ b/data/examples/declaration/value/function/pattern/as-pattern-out.hs
@@ -1,0 +1,3 @@
+main = case [1] of
+  xs@(x : _) -> print (x, xs)
+  xs@[] -> print xs

--- a/data/examples/declaration/value/function/pattern/as-pattern.hs
+++ b/data/examples/declaration/value/function/pattern/as-pattern.hs
@@ -1,0 +1,3 @@
+main = case [1] of
+  xs @ (x:_) -> print (x, xs)
+  xs@[] -> print xs

--- a/src/Ormolu/Parser.hs
+++ b/src/Ormolu/Parser.hs
@@ -82,6 +82,7 @@ manualExts =
   , TransformListComp -- steals the group keyword
   , UnboxedTuples -- breaks (#) lens operator
   , MagicHash -- screws {-# these things #-}
+  , TypeApplications -- steals (@) operator on some cases
   , AlternativeLayoutRule
   , AlternativeLayoutRuleTransitional
   , MonadComprehensions

